src/main/java/org/jboss/aesh/io/FileResource.java

```diff
@@ -104,6 +105,12 @@ public boolean delete() {
     }
 
     @Override
+    public void move(Resource target) throws IOException {
+        FileResource destination = (FileResource) target;
+        Files.move(this.file.toPath(), destination.file.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
I think that this should be an option of the client.. if they want to override or not. IMHO.